Core/PrivatTransactionsImporter.cs

```diff
@@ -9,72 +8,88 @@
-            };
-            var res = await client.SendAsync(request);
+			var client = new HttpClient();
+			var request = new HttpRequestMessage
+			{
+				Method = HttpMethod.Get,
+				RequestUri = new Uri("https://api.privatbank.ua/p24api/rest_fiz"),
+				Content = new StringContent(
+					$"<?xml version=\"1.0\" encoding=\"UTF-8\"?><request version=\"1.0\"><merchant><id>{credentials.MerchantId}</id><signature>{signature}</signature></merchant><data>{data}</data></request>",
+					Encoding.Default,
+					"application/xml")
+			};
+			var res = await client.SendAsync(request);
 
-            var xml = await res.Content.ReadAsStringAsync();
+			var xml = await res.Content.ReadAsStringAsync();
 
-            XmlDocument xmlDoc = new XmlDocument();
-            xmlDoc.LoadXml(xml);
+			var xmlDoc = new XmlDocument();
+			xmlDoc.LoadXml(xml);
 
-            string xpath = "response/data/info/statements/statement";
-            var nodes = xmlDoc.SelectNodes(xpath);
-            
-            var list = new List<Transaction>();
-            foreach (XmlNode childrenNode in nodes)
-            {
-                list.Add(GetTransactionFromXml(childrenNode));
-            }
-            return list;
-        }
+			const string xpath = "response/data/info/statements/statement";
+			var nodes = xmlDoc.SelectNodes(xpath);
 
-        private static string ByteArrayToString(byte[] ba)
-        {
-            StringBuilder hex = new StringBuilder(ba.Length * 2);
-            foreach (byte b in ba)
-                hex.AppendFormat("{0:x2}", b);
-            return hex.ToString();
-        }
+			var list = new List<Transaction>();
+			foreach (XmlNode childrenNode in nodes)
+			{
+				list.Add(GetTransactionFromXml(childrenNode));
+			}
 
-        private static Transaction GetTransactionFromXml(XmlNode node)
-        {
-            return new Transaction(node.Attributes["card"].Value,
-                node.Attributes["appcode"].Value,
-                DateTimeOffset.Parse(node.Attributes["trandate"].Value + " " + node.Attributes["trantime"].Value),
-                new Money(decimal.Parse(node.Attributes["cardamount"].Value.Split(' ').First()), (Currency)Enum.Parse(typeof(Currency), node.Attributes["cardamount"].Value.Split(' ').Last())),
-                new Money(decimal.Parse(node.Attributes["rest"].Value.Split(' ').First()), (Currency)Enum.Parse(typeof(Currency), node.Attributes["rest"].Value.Split(' ').Last())),
-                node.Attributes["terminal"].Value,
-                node.Attributes["description"].Value);
-        }
-    }
+			return list;
+		}
+
+		private static string ByteArrayToString(byte[] ba)
+		{
+			var hex = new StringBuilder(ba.Length * 2);
+			foreach (var b in ba)
+			{
+				_ = hex.AppendFormat("{0:x2}", b);
+			}
+
+			return hex.ToString();
+		}
+
+		public static Transaction GetTransactionFromXml(XmlNode node)
+		{
+			var cartNumber = node.Attributes["card"].Value;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'node' before using it.](https://app.codacy.com/manual/siriak/MoneyManager/pullRequest?prid=4931949)